http/req/doc.go

```diff
@@ -0,0 +1,16 @@
+/*
+Package req provides ergonomics for handling an HTTP request.
+
+Package req provides a helper for parsing payloads in an HTTP request.
+It supports JSON-encoded payloads and payloads encoded in query parameters.
+In both cases, package req expects to parse payloads into a pointer to a struct.
+That struct ought to leverage the apporpriate struct tags for performing two tasks.
+First, matching keys in the payload to fields on the struct.
+Second, for validating the payload's data meets requirements.
+
+By leveraging req, handlers can get data out of an HTTP request into its application specific structs.
```

Comment:
imo would be nice to have example parser usage code in the doc (or wherever makes the most sense)

Reply:
Yeah great suggestion, I was thinking of going with [Go's examples](https://go.dev/blog/examples). Too simple?

Reply:
that's actually what I had in mind too